src/coprocessor/dag/rpn_expr/types/expr_builder.rs

```diff
@@ -423,115 +419,71 @@ fn extract_scalar_value_json(val: Vec<u8>) -> Result<ScalarValue> {
-    impl FnA {
-        fn call(
-            _ctx: &mut EvalContext,
-            _payload: RpnFnCallPayload<'_>,
-            _v: &Option<i64>,
-        ) -> Result<Option<Real>> {
-            unreachable!()
-        }
+    #[rpn_fn]
+    fn fn_a(_v: &Option<i64>) -> Result<Option<Real>> {
+        unreachable!()
     }
 
     /// An RPN function for test. It accepts 2 float arguments, returns int.
-    #[derive(Debug, Clone, Copy, RpnFunction)]
-    #[rpn_function(args = 2)]
-    struct FnB;
-
-    impl FnB {
-        fn call(
-            _ctx: &mut EvalContext,
-            _payload: RpnFnCallPayload<'_>,
-            _v1: &Option<Real>,
-            _v2: &Option<Real>,
-        ) -> Result<Option<i64>> {
-            unreachable!()
-        }
+    #[rpn_fn]
+    fn fn_b(_v1: &Option<Real>, _v2: &Option<Real>) -> Result<Option<i64>> {
+        unreachable!()
     }
 
     /// An RPN function for test. It accepts 3 int arguments, returns int.
-    #[derive(Debug, Clone, Copy, RpnFunction)]
-    #[rpn_function(args = 3)]
-    struct FnC;
-
-    impl FnC {
-        fn call(
-            _ctx: &mut EvalContext,
-            _payload: RpnFnCallPayload<'_>,
-            _v1: &Option<i64>,
-            _v2: &Option<i64>,
-            _v3: &Option<i64>,
-        ) -> Result<Option<i64>> {
-            unreachable!()
-        }
+    #[rpn_fn]
+    fn fn_c(_v1: &Option<i64>, _v2: &Option<i64>, _v3: &Option<i64>) -> Result<Option<i64>> {
+        unreachable!()
     }
 
     /// An RPN function for test. It accepts 3 float arguments, returns float.
-    #[derive(Debug, Clone, Copy, RpnFunction)]
-    #[rpn_function(args = 3)]
-    struct FnD;
-
-    impl FnD {
-        fn call(
-            _ctx: &mut EvalContext,
-            _payload: RpnFnCallPayload<'_>,
-            _v1: &Option<Real>,
-            _v2: &Option<Real>,
-            _v3: &Option<Real>,
-        ) -> Result<Option<Real>> {
-            unreachable!()
-        }
+    #[rpn_fn]
+    fn fn_d(_v1: &Option<Real>, _v2: &Option<Real>, _v3: &Option<Real>) -> Result<Option<Real>> {
+        unreachable!()
     }
 
     /// For testing `append_rpn_nodes_recursively`. It accepts protobuf function sig enum, which
-    /// cannot be modified by us in tests to support FnA ~ FnD. So let's just hard code some
+    /// cannot be modified by us in tests to support fn_a ~ fn_d. So let's just hard code some
     /// substitute.
-    fn fn_mapper(value: ScalarFuncSig, _children: &[Expr]) -> Result<Box<dyn RpnFunction>> {
-        // FnA: CastIntAsInt
-        // FnB: CastIntAsReal
-        // FnC: CastIntAsString
-        // FnD: CastIntAsDecimal
-        match value {
-            ScalarFuncSig::CastIntAsInt => Ok(Box::new(FnA)),
-            ScalarFuncSig::CastIntAsReal => Ok(Box::new(FnB)),
-            ScalarFuncSig::CastIntAsString => Ok(Box::new(FnC)),
-            ScalarFuncSig::CastIntAsDecimal => Ok(Box::new(FnD)),
+    fn fn_mapper(value: ScalarFuncSig, _children: &[Expr]) -> Result<RpnFn> {
```

Comment:
I feel like it is better to call `RpnFn` as `RpnFnMeta` and call `xxx_fn()` as `xxx_create_fn_meta()`. How do think?

Reply:
Yes. `RpnFnMeta` is a better name. But I'm afraid `xxx_create_fn_meta()` may be too long...

Reply:
We don't have many places that will invoke `xxx_create_fn_meta`, not so bad. 😄

Or.. let's use new! `xxx_new_meta` 🤨

Reply:
Actually, I don't want "new" or "create" so much because the function always returns the same object...What do you think of just `xxx_fn_meta`?

Reply:
looks good